tests/acceptance/30_custom_promise_types/multiline_insert.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+#
+# Sample custom promise type, uses cfengine.py library located in same dir.
+#
+# Use it in the policy like this:
+# promise agent multiline_insert
+# {
+#     interpreter => "/usr/bin/python3";
+#     path => "$(sys.inputdir)/multiline_insert.py";
+# }
+# bundle agent main
+# {
+#   multiline_insert:
+#     "/home/vagrant/output"
+#       lines => { "line1", "line2", "line3" };
+# }
+
+import os.path
+
+from cfengine import PromiseModule, ValidationError, Result
+
+
+def _file_content_is(fpath, contents):
+    if not os.path.exists(fpath):
+        return False
+
```

Comment:
I guess this is not necessary and using the exception handling below is sufficient and more "pythonic"? (Also avoids the race condition).